detector.go

```diff
@@ -95,46 +128,66 @@ func (d *Detector) detectOrder(order buildpack.Order, done, next []buildpack.Gro
 	return nil, nil, ErrFailedDetection
 }
 
-func (d *Detector) detectGroup(group buildpack.Group, done []buildpack.GroupBuildpack, wg *sync.WaitGroup) ([]buildpack.GroupBuildpack, []platform.BuildPlanEntry, error) {
-	for i, groupBp := range group.Group {
-		key := groupBp.String()
-		if hasID(done, groupBp.ID) {
+func (d *Detector) detectGroup(group buildpack.Group, done []buildpack.GroupElement, wg *sync.WaitGroup) ([]buildpack.GroupElement, []platform.BuildPlanEntry, error) {
+	for i, groupEl := range group.Group {
+		// Continue if element has already been processed.
+		if hasID(done, groupEl.ID) {
 			continue
 		}
 
-		bp, err := d.Store.Lookup(groupBp.ID, groupBp.Version)
+		// Resolve order if element is the order for extensions.
```

Comment:
Having a single "special" group element to represent the order for extensions seems like the cleanest way to do it. I also experimented with having a special ID that, when provided to the store, would return a descriptor for an extensions order (as suggested [here](https://github.com/buildpacks/lifecycle/pull/812#discussion_r818160288)) but that turned out to be messier IMO.